libraries/Microsoft.Bot.Connector/Authentication/MicrosoftAppCredentials.cs

```diff
@@ -77,7 +60,12 @@ public MicrosoftAppCredentials(string appId, string password)
         {
             this.MicrosoftAppId = appId;
             this.MicrosoftAppPassword = password;
-            this.TokenCacheKey = $"{MicrosoftAppId}-cache";
+
+            authenticator = new Lazy<AdalAuthenticator>(() =>
```

Comment:
Does this need to be Lazy<>? 

Why not just actually initialize it here, or at part of startup? Making it lazy means it's somewhat non-deterministic when/if it may run (and fail). It also slightly complicates all the calling code.

Reply:
... and every bot needs this pretty much right away, so it's not really a "Win" from a perf perspective. It just slows down the 1st call that much more.